expense/activities.py

```diff
@@ -0,0 +1,89 @@
+import httpx
+from temporalio import activity
+
+from expense import EXPENSE_SERVER_HOST_PORT
+
+
+@activity.defn
+async def create_expense_activity(expense_id: str) -> None:
+    if not expense_id:
+        raise ValueError("expense id is empty")
+
+    async with httpx.AsyncClient() as client:
+        response = await client.get(
+            f"{EXPENSE_SERVER_HOST_PORT}/create",
+            params={"is_api_call": "true", "id": expense_id},
+        )
+        response.raise_for_status()
```

Comment:
_Technically_ 4xx is probably a `ApplicationError` with `non_retryable=True`, but that's a bit pedantic. But with this setup, an activity that, say, has invalid auth will retry forever